openedx_events/enterprise/data.py

```diff
@@ -22,3 +25,71 @@ class SubsidyRedemption:
+    """
+    Attributes of an ``openedx_ledger.Reversal`` record.
+
+    A ``Reversal`` is a model that represents the "undo-ing" of a ``Transaction`` (see below). It's primarily
+    used within the domain of edX Enterprise for recording unenrollments and refunds of subsidized
+    enterprise enrollments.
+    https://github.com/openedx/openedx-ledger/blob/master/openedx_ledger/models.py
+
+    Arguments:
+        uuid (str): Primary identifier of the record.
+        created (datetime): When the record was created.
+        modified (datetime): When the record was last modified.
+        idempotency_key (str): Client-generated unique value to achieve idempotency of operations.
+        quantity (int): How many units of value this reversal represents (e.g. USD cents).
+        state (str): Current lifecyle state of the record, one of (created, pending, committed, failed).
+    """
+
+    uuid = attr.ib(type=UUID)
+    created = attr.ib(type=datetime)
+    modified = attr.ib(type=datetime)
+    idempotency_key = attr.ib(type=str)
+    quantity = attr.ib(type=int)
+    state = attr.ib(type=str)
+
+
+@attr.s(frozen=True)
+class LedgerTransaction:
+    """
+    Attributes of an ``openedx_ledger.Transaction`` record.
+
+    A ``Transaction`` is a model that represents value moving in or out of a ``Ledger``. It's primarily
+    used within the domain of edX Enterprise for recording the redemption of subsidized enrollments.
+    https://github.com/openedx/openedx-ledger/blob/master/openedx_ledger/models.py
+
+    Arguments:
+        uuid (UUID): Primary identifier of the Transaction.
+        created (datetime): When the record was created.
+        modified (datetime): When the record was last modified.
+        idempotency_key (str): Client-generated unique value to achieve idempotency of operations.
+        quantity (int): How many units of value this transaction represents (e.g. USD cents).
+        state (str): Current lifecyle state of the record, one of (created, pending, committed, failed).
+        ledger_uuid (UUID): The primary identifier of this Transaction's ledger object.
+        subsidy_access_policy_uuid (UUID): The primary identifier of the subsidy access policy for this transaction.
+        lms_user_id (int): The LMS user id of the user associated with this transaction.
+        content_key (CourseKey): The course (run) key associated with this transaction.
+        parent_content_key (str): The parent (just course, not run) key for the course key.
+        fulfillment_identifier (str): The identifier of the subsidized enrollment record for a learner,
+          generated durning enrollment.
+        reversal (LedgerTransactionReversal): Any reversal associated with this transaction.
+    """
+
+    uuid = attr.ib(type=UUID)
+    created = attr.ib(type=datetime)
+    modified = attr.ib(type=datetime)
+    idempotency_key = attr.ib(type=str)
+    quantity = attr.ib(type=int)
+    state = attr.ib(type=str)
```

Comment:
What do you think about doing:

```suggestion
@attr.s(frozen=True)
class BaseLedgerTransaction:
    uuid = attr.ib(type=UUID)
    created = attr.ib(type=datetime)
    modified = attr.ib(type=datetime)
    idempotency_key = attr.ib(type=str)
    quantity = attr.ib(type=int)
    state = attr.ib(type=str)
    
@attr.s(frozen=True)
class LedgerTransaction(BaseLedgerTransaction):

@attr.s(frozen=True)
class LedgerTransactionReversal(BaseLedgerTransaction):
...
```
I'm unsure about the naming, so you can change it for a better one.